webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceService.scala

```diff
@@ -109,7 +117,49 @@ class DataSourceService @Inject()(
     if (emptyDirs.nonEmpty) logger.warn(s"Empty organization dataset dirs: ${emptyDirs.mkString(", ")}")
   }
 
-  def handleUpload(id: DataSourceId, dataSetZip: File): Fox[Unit] = {
+  def handleUpload(uploadId: String,
+                   datasourceId: DataSourceId,
+                   resumableUploadInformation: ResumableUploadInformation,
+                   currentChunkNumber: Int,
+                   chunkFile: File): Fox[Unit] = {
+    val needsWrite = pendingUploadChunks.synchronized {
```

Comment:
I find it a bit hard to grasp what `needsWrite` expresses, especially since it is determined via the return value of `set.add`? What is the semantic of this return value? Could needsWrite be renamed to make its use clearer?